packages/react/src/primitives/shared/getStyleValue.ts

```diff
@@ -0,0 +1,32 @@
+import { WebTheme, isDesignToken } from '@aws-amplify/ui';
+import { getCSSVariableIfValueIsThemeKey } from './utils';
+
+interface GetStyleValueProps {
+  value: unknown;
+  propKey?: string;
+  tokens: WebTheme['tokens'];
+}
+
+/**
+ * This takes an unknown value, which could be a:
+ * - design token: `color={tokens.colors.font.primary}`
+ * - string, which could be a:
+ *   - theme key: `color='font.primary'`
+ *   - plain style: `color='red'`
+ * - or a number: `padding={10}`
+ * @param value
+ * @param key
+ * @param tokens tokens object from the Theme
+ * @returns a string or a number
```

Comment:
Nit: probably don't need the JSDoc `@` comments since the TS interface documents what is accepted and returned? I tend to prefer "what this function is for" only comments in TS, since everything else can quickly get out of date with the TS type source of truth.